script.js

```diff
@@ -5,13 +5,27 @@ jQuery(function () {
     if ($sitemapNavi.length === 0) return;
 
     var $mediaToggle = jQuery(
-        '<label><input type="checkbox" checked="checked">&nbsp;' +
+        '<label><input type="checkbox">&nbsp;' +
         LANG.plugins.sitemapnavi.medialabel +
         '</label>'
     );
-    $mediaToggle.change(function () {
-        $sitemapNavi.find('li.media').toggle($mediaToggle.checked);
+
+    $mediaToggle.find('input').prop('checked', !$sitemapNavi.hasClass('hide-media-links'));
+
+    $mediaToggle.change(function (event) {
+        $sitemapNavi.toggleClass('hide-media-links');
+        DokuCookie.setValue('plugin_sitemapnavi_showmedia', event.target.checked);
     });
+
+    // update initially rendered default state if it does not match the value stored in cookie
+    if (
+        typeof (DokuCookie.getValue('plugin_sitemapnavi_showmedia')) !== 'undefined' &&
+        (DokuCookie.getValue('plugin_sitemapnavi_showmedia') === 'true') === $sitemapNavi.hasClass('hide-media-links')
```

Comment:
to avoid type problems with the cookie (also related to https://github.com/splitbrain/dokuwiki/pull/2722) could we always cast the value to integer?

Reply:
ok